webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/OrderedScrambles.java

```diff
@@ -0,0 +1,120 @@
+        }
+        
+        azzert(wcifHelper.getAllScrambleRequests() != null, "There should be scramble requests.");
+
+        boolean hasMultipleDays = wcifHelper.hasMultipleDays();
+        boolean hasMultipleVenues = wcifHelper.hasMultipleVenues();
+        
+        // We consider the competition start date as the earlier activity from the schedule.
+        // This prevents miscalculation of dates for multiple timezones.
+        String competitionStartString = wcifHelper.getEarlierActivityString();
+        
+        for (JsonElement venue : wcifHelper.getVenues()) {
+            String venueName = wcifHelper.getSafeVenueName(venue);
+                        
+            DateTimeZone timezone = wcifHelper.getTimeZone(venue);
+            DateTime competitionStartDate = new DateTime(competitionStartString, timezone);
+            
+            boolean hasMultipleRooms = wcifHelper.hasMultipleRooms(venue);
+
+            for (JsonElement room : wcifHelper.getRooms(venue)) {
+
+                ArrayList<Integer> dayList = new ArrayList<Integer>();
+                ArrayList<ArrayList<ScrambleRequest>> scrambleRequestListByDay = new ArrayList<ArrayList<ScrambleRequest>>();
+                
+                String roomName = wcifHelper.getSafeRoomName(room);
+                                
+                for (JsonElement activity : wcifHelper.getActivities(room)) {
+                    String activityCode = wcifHelper.getActivityCode(activity);
+
+                    for (ScrambleRequest scrambleRequest : wcifHelper.getScrambleRequests(activityCode)) {
+
+                        DateTime activityStartTime = wcifHelper.getActivityStartTime(activity, timezone);
+
+                        int activityDay = Days.daysBetween(competitionStartDate.withTimeAtStartOfDay(), activityStartTime.withTimeAtStartOfDay()).getDays()+1;
+                        
+                        if (!dayList.contains(activityDay)) {
+                            dayList.add(activityDay);
+                            scrambleRequestListByDay.add(new ArrayList<ScrambleRequest>());
+                        }
+
+                        int index = dayList.indexOf(activityDay);
+                        scrambleRequest.roundStartTime = activityStartTime;
+                        scrambleRequestListByDay.get(index).add(scrambleRequest);
+                    }
+                }
+
+                for (int index = 0; index<dayList.size(); index++) {
+                    if (scrambleRequestListByDay.get(index).size()>0) {
+                        
+                        Collections.sort(scrambleRequestListByDay.get(index));
+                        
+                        String pdfFileName = "Printing/Ordered Scrambles/";
+
+                        if (hasMultipleVenues) {
+                            pdfFileName += venueName+"/";
+                        }
+
+                        if (hasMultipleDays || dayList.size() > 1) { // Double check, just in case.
```

Comment:
I think it makes more sense to move this logic up into the line of code where we first declare `hasMultipleDays`. That would save you from having to do the same check down on line 100.

I also think that when you move this up, you should add a better comment. This is a pretty tricky situation that you had to do a bit of investigation to understand, and I think this is a good opportunity to teach future code readers about what you learned.

Reply:
Fixed